pkg/plugins/discovery/k8s/util/util.go

```diff
@@ -17,11 +17,28 @@ func MatchServiceThatSelectsPod(pod *kube_core.Pod) ServicePredicate {
 	}
 }
 
-func FindServices(svcs *kube_core.ServiceList, predicate ServicePredicate) []*kube_core.Service {
+// According to K8S docs about Service#selector:
+// Route service traffic to pods with label keys and values matching this selector.
+// If empty or not present, the service is assumed to have an external process managing its endpoints, which Kubernetes will not modify.
+// Only applies to types ClusterIP, NodePort, and LoadBalancer. Ignored if type is ExternalName. More info: https://kubernetes.io/docs/concepts/services-networking/service/
+//
+// When converting Pod to Dataplane, we don't want to take into account Services that has no Selector, otherwise any Pod will match this service
+// and since we just take any int target port in #util.FindPort every Dataplane in the same namespace as this service would get an extra inbound for it.
+func AnySelector() ServicePredicate {
+	return func(svc *kube_core.Service) bool {
+		return len(svc.Spec.Selector) > 0
+	}
+}
+
+func FindServices(svcs *kube_core.ServiceList, predicates ...ServicePredicate) []*kube_core.Service {
```

Comment:
Does using `FindServices` without `AnySelector()` predicate, make any sense?

Reply:
Not currently. Right now the only usage is
```
util_k8s.FindServices(allServices, util_k8s.AnySelector(), util_k8s.MatchServiceThatSelectsPod(pod))
```

Reply:
Yeah, maybe ignoring services without selectors should be a default behavior for `FindServices`?

Reply:
`FindServices` (for better or worse) is a generic function that is essentially a `filter` from languages that are not lacking in features. I don't think it's ok to assume in this function that it's ok to ignore selectors. This would have to be changed to `FindMatchingServices(pod)`